File: backend/src/main/java/com/locavia/backend/service/AvisReclamation/AvisService.java
package com.locavia.backend.service.AvisReclamation;

import com.locavia.backend.dto.AvisDTO;

import java.util.List;
import java.util.Map;

public interface AvisService {

    List<AvisDTO> getAll();

    AvisDTO getById(Long id);

    AvisDTO create(AvisDTO dto);

    AvisDTO update(Long id, AvisDTO dto);

    void delete(Long id);

    Map<String, Object> getStats();
}

File: frontend/src/app/core/services/avis.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Avis, AvisStats } from '../models/avis.model';

@Injectable({ providedIn: 'root' })
export class AvisService {
  private readonly apiUrl = `${environment.apiUrl}/api/avis`;

  constructor(private http: HttpClient) {}

  getAll(): Observable<Avis[]> {
    return this.http.get<Avis[]>(this.apiUrl);
  }

  getById(id: number): Observable<Avis> {
    return this.http.get<Avis>(`${this.apiUrl}/${id}`);
  }

  create(avis: Avis): Observable<Avis> {
    return this.http.post<Avis>(this.apiUrl, avis);
  }

  update(id: number, avis: Avis): Observable<Avis> {
    return this.http.put<Avis>(`${this.apiUrl}/${id}`, avis);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  getStats(): Observable<AvisStats> {
    return this.http.get<AvisStats>(`${this.apiUrl}/stats`);
  }
}
